src/controller/BridgeController.js

```diff
@@ -0,0 +1,136 @@
+const GameController = require('./GameController');
+// inputCheck
+const BridgeSize = require('../inputCheck/BridgeSize');
+const Moving = require('../inputCheck/Moving');
+const GameCommand = require('../inputCheck/GameCommand');
+// constants
+const { INPUT_VIEW } = require('../Constants');
+const { Console } = require('@woowacourse/mission-utils');
+
+const BridgeController = class extends GameController {
+  constructor(inputView, outputView, bridgeGame) {
+    super(inputView, outputView, bridgeGame);
+  }
+
+  runProcess() {
+    this.outputView.printStartMessage();
+    this.inputBridgeSize();
+  }
+
+  inputBridgeSize() {
+    const callbackInputBridgeSize = (size) => {
+      this.checkBridgeSize(size);
+      this.saveSize(size);
+      this.prebuildBridge(size);
+      this.inputMoving();
+    };
+
+    this.inputView.readBridgeSize(INPUT_VIEW.size_message, callbackInputBridgeSize);
+  }
+
+  checkBridgeSize(input, confirmedInput = null) {
+    try {
+      const bridgeSize = new BridgeSize(input);
+      confirmedInput = bridgeSize.checkInput();
+    } catch (error) {
+      this.outputView.printError(error.message);
+    } finally {
+      if (confirmedInput !== input) return this.inputBridgeSize();
+    }
+  }
+
+  saveSize(input) {
+    this.bridgeGame.saveSize(input);
+  }
+
+  prebuildBridge(input) {
+    this.bridgeGame.precompose(input);
+  }
+
+  inputMoving() {
+    const callbackInputMoving = (moving) => {
+      this.checkMoving(moving);
+    };
+
+    this.inputView.readMoving(INPUT_VIEW.moving_message, callbackInputMoving);
+  }
+
+  checkMoving(input, confirmedInput = null) {
+    try {
+      confirmedInput = this.checkMovingInput(input);
+    } catch (error) {
+      this.outputView.printError(error.message);
+    } finally {
+      if (confirmedInput !== input) return this.inputMoving();
+      return this.saveMoving(input);
+    }
+  }
```

Comment:
```suggestion
checkMoving(input) {
    try {
      this.checkMovingInput(input);
    } catch (error) {
      this.outputView.printError(error.message);
      this.inputMoving();
    }
    this.saveMoving(input);
  }
```

불필요한 변수와 동작들을 줄일 수 있을 것 같습니다.

Reply:
```js
checkMoving(input, confirmedInput = null) {
    try {
      confirmedInput = this.checkMovingInput(input);
    } catch (error) {
      this.errorHandler(error, this.inputMoving.bind(this));
    }
    return this.saveMoving(input);
  }
```

다들 이 부분을 언급해주시는 군요 !  
저 역시 다른분들 피드백을 해드리면서 깨달았습니다
확실히 불필요한 동작들이 많은 것 같습니다. 
지금은 위 와 같이 수정해보았습니다.